middlewares/accesslog/logger.go

```diff
@@ -125,8 +166,25 @@ func (l *LogHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request, next h
 	core[RequestProtocol] = req.Proto
 	core[RequestLine] = fmt.Sprintf("%s %s %s", req.Method, urlCopyString, req.Proto)
 
-	core[ClientAddr] = req.RemoteAddr
-	core[ClientHost], core[ClientPort] = silentSplitHostPort(req.RemoteAddr)
+	host, port := silentSplitHostPort(req.RemoteAddr)
+
+	// Mask host's address if necessary
+	isIPv6 := strings.Contains(host, ":")
+
+	if l.maskIPv4 != nil && l.maskIPv6 != nil {
```

Comment:
```go
if len(l.maskIPv4) > 0 && len(l.maskIPv6) > 0 {
```

Reply:
len(nil) == 0 
https://golang.org/pkg/builtin/#len

Reply:
but `nil != []`

Reply:
Yes but in this case it won't make any difference? 
https://play.golang.org/p/UZUFhDfM5sF

Reply:
on slice we prefer use `len`

Reply:
There you go